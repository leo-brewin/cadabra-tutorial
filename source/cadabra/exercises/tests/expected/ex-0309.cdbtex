\chardef\thehashcode=\catcode`\#	 % save catcode of hash (probably 6)
\catcode`\#=11			             % set catcode for hash to be a letter (11)
\cdbtag{ex-0309.dgab}{%
\partial_{c}{g^{a b}} \rightarrow -g^{a e} g^{b f} \partial_{c}{g_{e f}}%
}
\cdbtag{ex-0309.Gamma}{%
\Gamma^{a}{}_{b c} \rightarrow \frac{1}{2}g^{a e} \left(\partial_{b}{g_{e c}}\discretionary{}{}{}+\partial_{c}{g_{b e}}\discretionary{}{}{}-\partial_{e}{g_{b c}}\right)%
}
\cdbtag{ex-0309.Rabcd}{%
R^{a}{}_{b c d} \rightarrow \partial_{c}{\Gamma^{a}{}_{b d}}\discretionary{}{}{}+\Gamma^{a}{}_{e c} \Gamma^{e}{}_{b d}\discretionary{}{}{}-\partial_{d}{\Gamma^{a}{}_{b c}}\discretionary{}{}{}-\Gamma^{a}{}_{e d} \Gamma^{e}{}_{b c}%
}
\cdbtag{ex-0309.101}{%
4R^{c}{}_{a c b}%
}
\cdbtag{ex-0309.102}{%
4\partial_{c}{\Gamma^{c}{}_{a b}}\discretionary{}{}{}+4\Gamma^{c}{}_{e c} \Gamma^{e}{}_{a b}\discretionary{}{}{}-4\partial_{b}{\Gamma^{c}{}_{a c}}\discretionary{}{}{}-4\Gamma^{c}{}_{e b} \Gamma^{e}{}_{a c}%
}
\cdbtag{ex-0309.103}{%
2\partial_{c}\left(g^{c e} \left(\partial_{a}{g_{e b}}\discretionary{}{}{}+\partial_{b}{g_{a e}}\discretionary{}{}{}-\partial_{e}{g_{a b}}\right)\right)\discretionary{}{}{}+g^{c d} \left(\partial_{e}{g_{d c}}\discretionary{}{}{}+\partial_{c}{g_{e d}}\discretionary{}{}{}-\partial_{d}{g_{e c}}\right) g^{e f} \left(\partial_{a}{g_{f b}}\discretionary{}{}{}+\partial_{b}{g_{a f}}\discretionary{}{}{}-\partial_{f}{g_{a b}}\right)\discretionary{}{}{}-2\partial_{b}\left(g^{c e} \left(\partial_{a}{g_{e c}}\discretionary{}{}{}+\partial_{c}{g_{a e}}\discretionary{}{}{}-\partial_{e}{g_{a c}}\right)\right)\discretionary{}{}{}-g^{c d} \left(\partial_{e}{g_{d b}}\discretionary{}{}{}+\partial_{b}{g_{e d}}\discretionary{}{}{}-\partial_{d}{g_{e b}}\right) g^{e f} \left(\partial_{a}{g_{f c}}\discretionary{}{}{}+\partial_{c}{g_{a f}}\discretionary{}{}{}-\partial_{f}{g_{a c}}\right)%
}
\cdbtag{ex-0309.104}{%
2\partial_{c}{g^{c e}} \left(\partial_{a}{g_{e b}}\discretionary{}{}{}+\partial_{b}{g_{a e}}\discretionary{}{}{}-\partial_{e}{g_{a b}}\right)\discretionary{}{}{}+2g^{c e} \partial_{c}\left(\partial_{a}{g_{e b}}\discretionary{}{}{}+\partial_{b}{g_{a e}}\discretionary{}{}{}-\partial_{e}{g_{a b}}\right)\discretionary{}{}{}+g^{c d} \left(\partial_{e}{g_{d c}}\discretionary{}{}{}+\partial_{c}{g_{e d}}\discretionary{}{}{}-\partial_{d}{g_{e c}}\right) g^{e f} \left(\partial_{a}{g_{f b}}\discretionary{}{}{}+\partial_{b}{g_{a f}}\discretionary{}{}{}-\partial_{f}{g_{a b}}\right)\discretionary{}{}{}-2\partial_{b}{g^{c e}} \left(\partial_{a}{g_{e c}}\discretionary{}{}{}+\partial_{c}{g_{a e}}\discretionary{}{}{}-\partial_{e}{g_{a c}}\right)\discretionary{}{}{}-2g^{c e} \partial_{b}\left(\partial_{a}{g_{e c}}\discretionary{}{}{}+\partial_{c}{g_{a e}}\discretionary{}{}{}-\partial_{e}{g_{a c}}\right)\discretionary{}{}{}-g^{c d} \left(\partial_{e}{g_{d b}}\discretionary{}{}{}+\partial_{b}{g_{e d}}\discretionary{}{}{}-\partial_{d}{g_{e b}}\right) g^{e f} \left(\partial_{a}{g_{f c}}\discretionary{}{}{}+\partial_{c}{g_{a f}}\discretionary{}{}{}-\partial_{f}{g_{a c}}\right)%
}
\cdbtag{ex-0309.105}{%
2\partial_{c}{g^{c e}} \partial_{a}{g_{e b}}\discretionary{}{}{}+2\partial_{c}{g^{c e}} \partial_{b}{g_{a e}}\discretionary{}{}{}-2\partial_{c}{g^{c e}} \partial_{e}{g_{a b}}\discretionary{}{}{}+2g^{c e} \partial_{c a}{g_{e b}}\discretionary{}{}{}+2g^{c e} \partial_{c b}{g_{a e}}\discretionary{}{}{}-2g^{c e} \partial_{c e}{g_{a b}}\discretionary{}{}{}+g^{c d} \partial_{e}{g_{d c}} g^{e f} \partial_{a}{g_{f b}}\discretionary{}{}{}+g^{c d} \partial_{e}{g_{d c}} g^{e f} \partial_{b}{g_{a f}}\discretionary{}{}{}-g^{c d} \partial_{e}{g_{d c}} g^{e f} \partial_{f}{g_{a b}}\discretionary{}{}{}+g^{c d} \partial_{c}{g_{e d}} g^{e f} \partial_{a}{g_{f b}}\discretionary{}{}{}+g^{c d} \partial_{c}{g_{e d}} g^{e f} \partial_{b}{g_{a f}}\discretionary{}{}{}-g^{c d} \partial_{c}{g_{e d}} g^{e f} \partial_{f}{g_{a b}}\discretionary{}{}{}-g^{c d} \partial_{d}{g_{e c}} g^{e f} \partial_{a}{g_{f b}}\discretionary{}{}{}-g^{c d} \partial_{d}{g_{e c}} g^{e f} \partial_{b}{g_{a f}}\discretionary{}{}{}+g^{c d} \partial_{d}{g_{e c}} g^{e f} \partial_{f}{g_{a b}}\discretionary{}{}{}-2\partial_{b}{g^{c e}} \partial_{a}{g_{e c}}\discretionary{}{}{}-2\partial_{b}{g^{c e}} \partial_{c}{g_{a e}}\discretionary{}{}{}+2\partial_{b}{g^{c e}} \partial_{e}{g_{a c}}\discretionary{}{}{}-2g^{c e} \partial_{b a}{g_{e c}}\discretionary{}{}{}%%
-2g^{c e} \partial_{b c}{g_{a e}}\discretionary{}{}{}+2g^{c e} \partial_{b e}{g_{a c}}\discretionary{}{}{}-g^{c d} \partial_{e}{g_{d b}} g^{e f} \partial_{a}{g_{f c}}\discretionary{}{}{}-g^{c d} \partial_{e}{g_{d b}} g^{e f} \partial_{c}{g_{a f}}\discretionary{}{}{}+g^{c d} \partial_{e}{g_{d b}} g^{e f} \partial_{f}{g_{a c}}\discretionary{}{}{}-g^{c d} \partial_{b}{g_{e d}} g^{e f} \partial_{a}{g_{f c}}\discretionary{}{}{}-g^{c d} \partial_{b}{g_{e d}} g^{e f} \partial_{c}{g_{a f}}\discretionary{}{}{}+g^{c d} \partial_{b}{g_{e d}} g^{e f} \partial_{f}{g_{a c}}\discretionary{}{}{}+g^{c d} \partial_{d}{g_{e b}} g^{e f} \partial_{a}{g_{f c}}\discretionary{}{}{}+g^{c d} \partial_{d}{g_{e b}} g^{e f} \partial_{c}{g_{a f}}\discretionary{}{}{}-g^{c d} \partial_{d}{g_{e b}} g^{e f} \partial_{f}{g_{a c}}%
}
\cdbtag{ex-0309.106}{%
-2g^{c d} g^{e f} \partial_{c}{g_{d f}} \partial_{a}{g_{e b}}\discretionary{}{}{}-2g^{c d} g^{e f} \partial_{c}{g_{d f}} \partial_{b}{g_{a e}}\discretionary{}{}{}+2g^{c d} g^{e f} \partial_{c}{g_{d f}} \partial_{e}{g_{a b}}\discretionary{}{}{}+2g^{c e} \partial_{c a}{g_{e b}}\discretionary{}{}{}+2g^{c e} \partial_{c b}{g_{a e}}\discretionary{}{}{}-2g^{c e} \partial_{c e}{g_{a b}}\discretionary{}{}{}+g^{c d} \partial_{e}{g_{d c}} g^{e f} \partial_{a}{g_{f b}}\discretionary{}{}{}+g^{c d} \partial_{e}{g_{d c}} g^{e f} \partial_{b}{g_{a f}}\discretionary{}{}{}-g^{c d} \partial_{e}{g_{d c}} g^{e f} \partial_{f}{g_{a b}}\discretionary{}{}{}+g^{c d} \partial_{c}{g_{e d}} g^{e f} \partial_{a}{g_{f b}}\discretionary{}{}{}+g^{c d} \partial_{c}{g_{e d}} g^{e f} \partial_{b}{g_{a f}}\discretionary{}{}{}-g^{c d} \partial_{c}{g_{e d}} g^{e f} \partial_{f}{g_{a b}}\discretionary{}{}{}-g^{c d} \partial_{d}{g_{e c}} g^{e f} \partial_{a}{g_{f b}}\discretionary{}{}{}-g^{c d} \partial_{d}{g_{e c}} g^{e f} \partial_{b}{g_{a f}}\discretionary{}{}{}+g^{c d} \partial_{d}{g_{e c}} g^{e f} \partial_{f}{g_{a b}}\discretionary{}{}{}+2g^{c d} g^{e f} \partial_{b}{g_{d f}} \partial_{a}{g_{e c}}\discretionary{}{}{}+2g^{c d} g^{e f} \partial_{b}{g_{d f}} \partial_{c}{g_{a e}}\discretionary{}{}{}-2g^{c d} g^{e f} \partial_{b}{g_{d f}} \partial_{e}{g_{a c}}\discretionary{}{}{}-2g^{c e} \partial_{b a}{g_{e c}}\discretionary{}{}{}%%
-2g^{c e} \partial_{b c}{g_{a e}}\discretionary{}{}{}+2g^{c e} \partial_{b e}{g_{a c}}\discretionary{}{}{}-g^{c d} \partial_{e}{g_{d b}} g^{e f} \partial_{a}{g_{f c}}\discretionary{}{}{}-g^{c d} \partial_{e}{g_{d b}} g^{e f} \partial_{c}{g_{a f}}\discretionary{}{}{}+g^{c d} \partial_{e}{g_{d b}} g^{e f} \partial_{f}{g_{a c}}\discretionary{}{}{}-g^{c d} \partial_{b}{g_{e d}} g^{e f} \partial_{a}{g_{f c}}\discretionary{}{}{}-g^{c d} \partial_{b}{g_{e d}} g^{e f} \partial_{c}{g_{a f}}\discretionary{}{}{}+g^{c d} \partial_{b}{g_{e d}} g^{e f} \partial_{f}{g_{a c}}\discretionary{}{}{}+g^{c d} \partial_{d}{g_{e b}} g^{e f} \partial_{a}{g_{f c}}\discretionary{}{}{}+g^{c d} \partial_{d}{g_{e b}} g^{e f} \partial_{c}{g_{a f}}\discretionary{}{}{}-g^{c d} \partial_{d}{g_{e b}} g^{e f} \partial_{f}{g_{a c}}%
}
\cdbtag{ex-0309.107}{%
-2\partial_{a}{g_{e b}} \partial_{c}{g_{d f}} g^{c d} g^{e f}\discretionary{}{}{}-2\partial_{b}{g_{a e}} \partial_{c}{g_{d f}} g^{c d} g^{e f}\discretionary{}{}{}+2\partial_{c}{g_{d f}} \partial_{e}{g_{a b}} g^{c d} g^{e f}\discretionary{}{}{}+2\partial_{c a}{g_{e b}} g^{c e}\discretionary{}{}{}+2\partial_{c b}{g_{a e}} g^{c e}\discretionary{}{}{}-2\partial_{c e}{g_{a b}} g^{c e}\discretionary{}{}{}+\partial_{a}{g_{f b}} \partial_{e}{g_{d c}} g^{c d} g^{e f}\discretionary{}{}{}+\partial_{b}{g_{a f}} \partial_{e}{g_{d c}} g^{c d} g^{e f}\discretionary{}{}{}-\partial_{e}{g_{d c}} \partial_{f}{g_{a b}} g^{c d} g^{e f}\discretionary{}{}{}+\partial_{a}{g_{f b}} \partial_{c}{g_{e d}} g^{c d} g^{e f}\discretionary{}{}{}+\partial_{b}{g_{a f}} \partial_{c}{g_{e d}} g^{c d} g^{e f}\discretionary{}{}{}-\partial_{c}{g_{e d}} \partial_{f}{g_{a b}} g^{c d} g^{e f}\discretionary{}{}{}-\partial_{a}{g_{f b}} \partial_{d}{g_{e c}} g^{c d} g^{e f}\discretionary{}{}{}-\partial_{b}{g_{a f}} \partial_{d}{g_{e c}} g^{c d} g^{e f}\discretionary{}{}{}+\partial_{d}{g_{e c}} \partial_{f}{g_{a b}} g^{c d} g^{e f}\discretionary{}{}{}+2\partial_{a}{g_{e c}} \partial_{b}{g_{d f}} g^{c d} g^{e f}\discretionary{}{}{}+2\partial_{b}{g_{d f}} \partial_{c}{g_{a e}} g^{c d} g^{e f}\discretionary{}{}{}-2\partial_{b}{g_{d f}} \partial_{e}{g_{a c}} g^{c d} g^{e f}\discretionary{}{}{}-2\partial_{b a}{g_{e c}} g^{c e}\discretionary{}{}{}%%
-2\partial_{b c}{g_{a e}} g^{c e}\discretionary{}{}{}+2\partial_{b e}{g_{a c}} g^{c e}\discretionary{}{}{}-\partial_{a}{g_{f c}} \partial_{e}{g_{d b}} g^{c d} g^{e f}\discretionary{}{}{}-\partial_{c}{g_{a f}} \partial_{e}{g_{d b}} g^{c d} g^{e f}\discretionary{}{}{}+\partial_{e}{g_{d b}} \partial_{f}{g_{a c}} g^{c d} g^{e f}\discretionary{}{}{}-\partial_{a}{g_{f c}} \partial_{b}{g_{e d}} g^{c d} g^{e f}\discretionary{}{}{}-\partial_{b}{g_{e d}} \partial_{c}{g_{a f}} g^{c d} g^{e f}\discretionary{}{}{}+\partial_{b}{g_{e d}} \partial_{f}{g_{a c}} g^{c d} g^{e f}\discretionary{}{}{}+\partial_{a}{g_{f c}} \partial_{d}{g_{e b}} g^{c d} g^{e f}\discretionary{}{}{}+\partial_{c}{g_{a f}} \partial_{d}{g_{e b}} g^{c d} g^{e f}\discretionary{}{}{}-\partial_{d}{g_{e b}} \partial_{f}{g_{a c}} g^{c d} g^{e f}%
}
\cdbtag{ex-0309.108}{%
-2\partial_{a}{g_{d b}} \partial_{c}{g_{e f}} g^{c e} g^{d f}\discretionary{}{}{}-2\partial_{b}{g_{a d}} \partial_{c}{g_{e f}} g^{c e} g^{d f}\discretionary{}{}{}+2\partial_{c}{g_{e f}} \partial_{d}{g_{a b}} g^{c e} g^{d f}\discretionary{}{}{}+2\partial_{c a}{g_{d b}} g^{c d}\discretionary{}{}{}+2\partial_{c b}{g_{a d}} g^{c d}\discretionary{}{}{}-2\partial_{c d}{g_{a b}} g^{c d}\discretionary{}{}{}+\partial_{a}{g_{d b}} \partial_{c}{g_{e f}} g^{f e} g^{c d}\discretionary{}{}{}+\partial_{b}{g_{a d}} \partial_{c}{g_{e f}} g^{f e} g^{c d}\discretionary{}{}{}-\partial_{c}{g_{e f}} \partial_{d}{g_{a b}} g^{f e} g^{c d}\discretionary{}{}{}+\partial_{a}{g_{d b}} \partial_{c}{g_{e f}} g^{c f} g^{e d}\discretionary{}{}{}+\partial_{b}{g_{a d}} \partial_{c}{g_{e f}} g^{c f} g^{e d}\discretionary{}{}{}-\partial_{c}{g_{e f}} \partial_{d}{g_{a b}} g^{c f} g^{e d}\discretionary{}{}{}-\partial_{a}{g_{d b}} \partial_{c}{g_{e f}} g^{f c} g^{e d}\discretionary{}{}{}-\partial_{b}{g_{a d}} \partial_{c}{g_{e f}} g^{f c} g^{e d}\discretionary{}{}{}+\partial_{c}{g_{e f}} \partial_{d}{g_{a b}} g^{f c} g^{e d}\discretionary{}{}{}+2\partial_{a}{g_{c d}} \partial_{b}{g_{e f}} g^{d e} g^{c f}\discretionary{}{}{}+2\partial_{b}{g_{d e}} \partial_{c}{g_{a f}} g^{c d} g^{f e}\discretionary{}{}{}-2\partial_{b}{g_{d e}} \partial_{c}{g_{a f}} g^{f d} g^{c e}\discretionary{}{}{}-2\partial_{b a}{g_{c d}} g^{d c}\discretionary{}{}{}%%
-2\partial_{b c}{g_{a d}} g^{c d}\discretionary{}{}{}+2\partial_{b c}{g_{a d}} g^{d c}\discretionary{}{}{}-\partial_{a}{g_{d e}} \partial_{c}{g_{f b}} g^{e f} g^{c d}\discretionary{}{}{}-\partial_{c}{g_{a e}} \partial_{d}{g_{f b}} g^{c f} g^{d e}\discretionary{}{}{}+\partial_{c}{g_{e b}} \partial_{d}{g_{a f}} g^{f e} g^{c d}\discretionary{}{}{}-\partial_{a}{g_{c d}} \partial_{b}{g_{e f}} g^{d f} g^{e c}\discretionary{}{}{}-\partial_{b}{g_{d e}} \partial_{c}{g_{a f}} g^{c e} g^{d f}\discretionary{}{}{}+\partial_{b}{g_{d e}} \partial_{c}{g_{a f}} g^{f e} g^{d c}\discretionary{}{}{}+\partial_{a}{g_{d e}} \partial_{c}{g_{f b}} g^{e c} g^{f d}\discretionary{}{}{}+\partial_{c}{g_{a e}} \partial_{d}{g_{f b}} g^{c d} g^{f e}\discretionary{}{}{}-\partial_{c}{g_{e b}} \partial_{d}{g_{a f}} g^{f c} g^{e d}%
}
\cdbtag{ex-0309.109}{%
-2\partial_{a}{g_{b c}} \partial_{d}{g_{e f}} g^{c e} g^{d f}\discretionary{}{}{}-2\partial_{b}{g_{a c}} \partial_{d}{g_{e f}} g^{c e} g^{d f}\discretionary{}{}{}+2\partial_{c}{g_{a b}} \partial_{d}{g_{e f}} g^{c e} g^{d f}\discretionary{}{}{}+2\partial_{a c}{g_{b d}} g^{c d}\discretionary{}{}{}+2\partial_{b c}{g_{a d}} g^{c d}\discretionary{}{}{}-2\partial_{c d}{g_{a b}} g^{c d}\discretionary{}{}{}+\partial_{a}{g_{b c}} \partial_{d}{g_{e f}} g^{c d} g^{e f}\discretionary{}{}{}+\partial_{b}{g_{a c}} \partial_{d}{g_{e f}} g^{c d} g^{e f}\discretionary{}{}{}-\partial_{c}{g_{a b}} \partial_{d}{g_{e f}} g^{c d} g^{e f}\discretionary{}{}{}+\partial_{a}{g_{c d}} \partial_{b}{g_{e f}} g^{c e} g^{d f}\discretionary{}{}{}-2\partial_{a b}{g_{c d}} g^{c d}\discretionary{}{}{}-2\partial_{c}{g_{a d}} \partial_{e}{g_{b f}} g^{c f} g^{d e}\discretionary{}{}{}+2\partial_{c}{g_{a d}} \partial_{e}{g_{b f}} g^{c e} g^{d f}%
}
\cdbtag{ex-0309.110}{%
-2g^{c d} g^{e f} \partial_{a}{g_{b c}} \partial_{e}{g_{d f}}\discretionary{}{}{}-2g^{c d} g^{e f} \partial_{b}{g_{a c}} \partial_{e}{g_{d f}}\discretionary{}{}{}+2g^{c d} g^{e f} \partial_{c}{g_{a b}} \partial_{e}{g_{d f}}\discretionary{}{}{}+2g^{c d} \partial_{a c}{g_{b d}}\discretionary{}{}{}+2g^{c d} \partial_{b c}{g_{a d}}\discretionary{}{}{}-2g^{c d} \partial_{c d}{g_{a b}}\discretionary{}{}{}+g^{c d} g^{e f} \partial_{a}{g_{b c}} \partial_{d}{g_{e f}}\discretionary{}{}{}+g^{c d} g^{e f} \partial_{b}{g_{a c}} \partial_{d}{g_{e f}}\discretionary{}{}{}-g^{c d} g^{e f} \partial_{c}{g_{a b}} \partial_{d}{g_{e f}}\discretionary{}{}{}+g^{c d} g^{e f} \partial_{a}{g_{c e}} \partial_{b}{g_{d f}}\discretionary{}{}{}-2g^{c d} \partial_{a b}{g_{c d}}\discretionary{}{}{}-2g^{c d} g^{e f} \partial_{c}{g_{a e}} \partial_{f}{g_{b d}}\discretionary{}{}{}+2g^{c d} g^{e f} \partial_{c}{g_{a e}} \partial_{d}{g_{b f}}%
}
\catcode`\#=\thehashcode          % restore original catcode for hash
