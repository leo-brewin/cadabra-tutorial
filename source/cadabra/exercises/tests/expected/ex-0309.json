{
  "FourRab": "-2g^{c d} g^{e f} \\partial_{a}(g_{b c}) \\partial_{e}(g_{d f})-2g^{c d} g^{e f} \\partial_{b}(g_{a c}) \\partial_{e}(g_{d f}) + 2g^{c d} g^{e f} \\partial_{c}(g_{a b}) \\partial_{e}(g_{d f}) + 2g^{c d} \\partial_{a c}(g_{b d}) + 2g^{c d} \\partial_{b c}(g_{a d})-2g^{c d} \\partial_{c d}(g_{a b}) + g^{c d} g^{e f} \\partial_{a}(g_{b c}) \\partial_{d}(g_{e f}) + g^{c d} g^{e f} \\partial_{b}(g_{a c}) \\partial_{d}(g_{e f})-g^{c d} g^{e f} \\partial_{c}(g_{a b}) \\partial_{d}(g_{e f}) + g^{c d} g^{e f} \\partial_{a}(g_{c e}) \\partial_{b}(g_{d f})-2g^{c d} \\partial_{a b}(g_{c d})-2g^{c d} g^{e f} \\partial_{c}(g_{a e}) \\partial_{f}(g_{b d}) + 2g^{c d} g^{e f} \\partial_{c}(g_{a e}) \\partial_{d}(g_{b f})"
}
